ethereumj-core/src/main/java/org/ethereum/vm/DataWord.java

```diff
@@ -22,77 +22,118 @@
-    public static final DataWord ZERO_EMPTY_ARRAY = new DataWord(new byte[0]);      // don't push it in to the stack
+    public static final DataWord ZERO = new DataWord(new byte[32]);
+    public static final DataWord ONE = DataWord.of((byte) 1);
 
     public static final long MEM_SIZE = 32 + 16 + 16;
 
-    private byte[] data = new byte[32];
+    private final byte[] data;
 
-    public DataWord() {
+    /**
+     * Unsafe private constructor
+     * Doesn't guarantee immutability if byte[] contents are changed later
+     * Use one of factory methods instead:
+     *  - {@link #of(byte[])}
+     *  - {@link #of(ByteArrayWrapper)}
+     *  - {@link #of(String)}
+     *  - {@link #of(long)}
+     *  - {@link #of(int)}
+     * @param data Byte Array[32] which is guaranteed to be immutable
+     */
+    private DataWord(byte[] data) {
+        if (data == null || data.length != 32) throw new RuntimeException("Input byte array should have 32 bytes in it!");
+        this.data = data;
     }
 
-    public DataWord(int num) {
-        this(ByteBuffer.allocate(4).putInt(num));
-    }
+    public static DataWord of(byte[] data) {
+        if (data == null || data.length == 0) {
+            return DataWord.ZERO;
+        }
+
+        int leadingZeroBits = numberOfLeadingZeros(data);
+        int valueBits = 8 * data.length - leadingZeroBits;
+        if (valueBits <= 8) {
+            if (data[data.length - 1] == 0) return DataWord.ZERO;
+            if (data[data.length - 1] == 1) return DataWord.ONE;
+        }
 
-    public DataWord(long num) {
-        this(ByteBuffer.allocate(8).putLong(num));
+        if (data.length == 32)
+            return new DataWord(Arrays.copyOf(data, data.length));
+        else if (data.length <= 32) {
+            byte[] bytes = new byte[32];
+            System.arraycopy(data, 0, bytes, 32 - data.length, data.length);
+            return new DataWord(bytes);
+        } else {
+            throw new RuntimeException(String.format("Data word can't exceed 32 bytes: 0x%s", ByteUtil.toHexString(data)));
+        }
     }
 
-    private DataWord(ByteBuffer buffer) {
-        final ByteBuffer data = ByteBuffer.allocate(32);
-        final byte[] array = buffer.array();
-        System.arraycopy(array, 0, data.array(), 32 - array.length, array.length);
-        this.data = data.array();
+    public static DataWord of(ByteArrayWrapper wrappedData) {
+        return of(wrappedData.getData());
     }
 
     @JsonCreator
-    public DataWord(String data) {
-        this(Hex.decode(data));
+    public static DataWord of(String data) {
+        return of(Hex.decode(data));
+    }
+
+    public static DataWord of(byte num) {
+        byte[] bb = new byte[32];
+        bb[31] = num;
+        return new DataWord(bb);
+
     }
 
-    public DataWord(ByteArrayWrapper wrappedData){
-        this(wrappedData.getData());
+    public static DataWord of(int num) {
+        return of(ByteBuffer.allocate(4).putInt(num).array());
```

Comment:
It's better use `Integer.BYTES` constant.

Reply:
We've got this already: `org.ethereum.util.ByteUtil.longToBytes`